Automattic/jetpack
# Boost: add full page caching

A WIP snapshot of what this code looks like.

This Boost module will create cached files from requests and place them in wp-content/boost-cache/cache/
It avoids caching POST and "backend" requests like wp-admin/ajax/rest.
It uses the cookie as the cache key. Cache files are stored in separate directories so it would be possible to track where one user goes.

No garbage collection, no effort to delete cache files when a post is updated.

Fixes #35042

### Other information:


## Jetpack product discussion
pc9hqz-2rA-p2

## Does this pull request change what data or activity we track or use?
No

## Testing instructions:
* Install Boost and apply this PR.
* Define the constant BOOST_CACHE = true in wp-config.php
* Visit the Boost settings page in wp-admin which will fire off the code that creates wp-content/advanced-cache.php and adds the WP_CACHE define to wp-config.php
* Look in wp-content/boost-cache/cache/ for new directories and files. `ls -lR` will be useful there as directories are deep.
* View the source of a page, scroll to the end, and you should see the text, "&lt;!-- cached -->" if the page was served from cache.
